### PR TITLE
Add copy button to highlight bar

### PR DESCRIPTION
**Pull request by 340rahul**
Addresses issue [2267](https://github.com/omnivore-app/omnivore/issues/2267)

- Adds a copy button to the highlight bar to allow user to copy selected/highlighted text to clipboard
- Clean up of `HighlightBar.tsx` to remove dup code

### CODE REVIEW

File: packages/web/components/templates/article/HighlightsLayer.tsx [hunk 1]
```diff
@@ -454,6 +455,30 @@ export function HighlightsLayer(props: HighlightsLayerProps): JSX.Element {
...
+              focusedHighlight.id
+            )
+            highlightedElements.forEach(
+              (element) => (textToCopy += element.textContent)
+            )
+          } else if (userSelectionText) {
+            textToCopy = userSelectionText
+          }
+
+          if (textToCopy) await navigator.clipboard.writeText(textToCopy)
```
Comment (jacksonh) [inline packages/web/components/templates/article/HighlightsLayer.tsx:476]: We might want to add a success toast

Comment (340rahul) [reply packages/web/components/templates/article/HighlightsLayer.tsx:476]: I guess we will want to use the `showSuccessToast` from `toastHelpers.tsx` for this? Is "Copied Successfully!" a good message? Might be a corner case (if user has the clipboard access disabled) but do we want to show an error toast as well?

Comment (jacksonh) [reply packages/web/components/templates/article/HighlightsLayer.tsx:476]: Yeah we can do an error toast too, something like:

```
if (success) {
   showSuccessToast('Highlight copied', {
            position: 'bottom-right',
          })
   } else {
   showErrorToast('Error copying highlight, permission denied.',  {
            position: 'bottom-right',
          })
```
   }

Comment (340rahul) [reply packages/web/components/templates/article/HighlightsLayer.tsx]: Added the toasts :)

**Comment by jacksonh** [comment]
Thanks! Took on the scariest file in our code base too. Going to get this into our staging environment and hopefully on to production later today or tomorrow (we have a few web things stacked up on staging right now that I want to test more)

**Comment by jacksonh** [comment]
Will have to tweak this a bit for mobile

IMG_5CB83E977E46-1

**Comment by 340rahul** [comment]
> Will have to tweak this a bit for mobile
> 
> IMG_5CB83E977E46-1

Oops! Sorry, I missed this. Working on a fix.

**Pull request merged**